src/dxvk/dxvk_fence.cpp

```diff
@@ -56,28 +56,55 @@ namespace dxvk {
         Logger::warn(str::format("Importing semaphores of type ", info.sharedType, " not supported by device"));
       }
     }
-
-    m_thread = dxvk::thread([this] { run(); });
   }
 
 
   DxvkFence::~DxvkFence() {
-    m_stop.store(true);
-    m_thread.join();
+    std::unique_lock<dxvk::mutex> lock(m_mutex);
+    bool threadRunning = m_running;
```

Comment:
this could be simplified quite a bit, just do something like
```c++
if (m_thread.joinable()) {
  { std::unique_lock /* ... */
    // set flag, notify cond
  }
  m_thread.join();
}
```